### PR TITLE
attempt fixing TypeError in Job

### PR DESCRIPTION
**Pull request by humzashah**
While unpacking a message, we're getting an empty Hash part which
indicates a garbled message. This PR ensures that an error isn't raised
when this happens and that we log the details as a warning. This aims to
fix the Routemaster crashes we've recently seen.

**Screenshots of error being seen on New Relic:**

screenshot from 2018-04-10 11-14-39

screenshot from 2018-04-10 11-14-48

**Screenshot of log from staging after these changes:**

screenshot from 2018-04-10 11-22-28

**Screenshot showing the number of ECS worker crashes before and after this change (the reduction is after this change):**

screenshot from 2018-04-10 11-28-23

### CODE REVIEW

File: routemaster/services/codec.rb [hunk 1]
```diff
@@ -19,6 +19,10 @@ def load(data)
         _log.warn { "Failed to decode message" }
         _log_exception(e)
         Models::Message::Garbled.new
+      rescue TypeError => e
+        _log.warn { "Failed decoding message. Code #{code}, Data #{data}, Hash #{hash}" }
```
Comment (gurkanoluc) [inline routemaster/services/codec.rb]: Nit: If we can use code=code, data=data format, until we move to logging with JSON those fields will be parsed by Datadog.

**Review by gurkanoluc — APPROVED** [review]
LGTM with a nit.

File: routemaster/services/codec.rb (hunk 1 above)
Comment (humzashah) [reply routemaster/services/codec.rb]: Sure, does this look better?

`"Failed decoding message. code=#{code}, data=#{data}, hash=#{hash}"`

Comment (gurkanoluc) [reply routemaster/services/codec.rb]: Image

Comment (humzashah) [reply routemaster/services/codec.rb]: Strange choice of GIF, TBH.

**Review by marcinwyszynski — APPROVED** [review]
If we didn't insist that NewRelic is all we need for exception tracking, we'd have caught this ages ago.

File: routemaster/services/codec.rb (hunk 1 above)
Comment (gurkanoluc) [reply routemaster/services/codec.rb]: You know, Giphy.

**Comment by humzashah** [comment]
@marcinwyszynski agreed - we need to figure out which CloudWatch metrics we should alert teams on.

**Comment by marcinwyszynski** [comment]
I mostly meant we should use Sentry or the like.

**Comment by humzashah** [comment]
@marcinwyszynski could you elaborate on why you think Sentry would've helped?

**Comment by marcinwyszynski** [comment]
Normally tools like Sentry allow you to set up email notifications when a previously unseen error occurs. You can also set up an integration with something like Jira to turn those into repair items. This is a very effective way of dealing with software failures. I'm personally a big fan of letting things crash, as long as I immediately get a stack trace in front of my eyes.

**Comment by humzashah** [comment]
Ah! Thanks for explaining, @marcinwyszynski. :bowing_man:

**Comment by pedrocunha** [comment]
Very true. One of the reasons we use Sentry in all our systems are notified in slack channels whenever a new exception arises for all logistics services

On 10 Apr 2018, 11:45 +0100, Syed Humza Shah <notifications@github.com>, wrote:
> Ah! Thanks for explaining, @marcinwyszynski. 🙇
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub, or mute the thread.

-- 




This email, including any attachments, is confidential and may be 
privileged. If you are not the intended recipient please notify the sender 
immediately, and please delete it; you should not copy it or use it for any 
purpose or disclose its contents to any other person. If the email, 
including any attachments, is legally privileged, you should not forward it 
to any other person.

**Pull request merged**